src/models/subscriptiondata.cpp

```diff
@@ -161,6 +162,15 @@ bool SubscriptionData::fromJson(const QByteArray& json) {
       return false;
     }
 
+    QJsonArray subscriptionsList = paymentData["subscriptions"].toArray();
+    for (int i = 0; i < subscriptionsList.count(); i++) {
```

Comment:
This can be also written in this way:

```
for (QJsonValue subscriptionValue : subscriptionsList) {
  QJsonObject subscription = subscriptionValue.toObject();
```